db/structure.sql

```diff
@@ -12,79 +12,261 @@ SET client_encoding = 'UTF8';
+		END IF;
+
+		-- do not execute if superuser
+		IF EXISTS (SELECT 1 FROM pg_user WHERE usename = current_user and usesuper = true) THEN
+			RETURN;
+		END IF;
+
+		-- do not execute if member of manager role
+		IF pg_has_role(current_user, 'rdsbroker_cbcf231f_9fac_4fa5_8551_85d8d425cf4a_manager', 'member') THEN
+			RETURN;
+		END IF;
+
+		IF pg_has_role(current_user, 'rdsbroker_cbcf231f_9fac_4fa5_8551_85d8d425cf4a_reader', 'member') THEN
+			RAISE EXCEPTION 'executing % is disabled for read only bindings', tg_tag;
+		END IF;
+	end
+$$;
 
 
 --
--- Name: EXTENSION pg_trgm; Type: COMMENT; Schema: -; Owner: -
+-- Name: make_readable(); Type: FUNCTION; Schema: public; Owner: -
 --
 
-COMMENT ON EXTENSION pg_trgm IS 'text similarity measurement and index searching based on trigrams';
+CREATE FUNCTION public.make_readable() RETURNS event_trigger
+    LANGUAGE plpgsql
+    SET search_path TO 'public'
+    AS $$
+	begin
+		IF EXISTS (SELECT 1 FROM pg_event_trigger_ddl_commands() WHERE schema_name NOT LIKE 'pg_temp%') THEN
+			EXECUTE 'select make_readable_generic()';
+			RETURN;
+		END IF;
+	end
+	$$;
 
 
 --
--- Name: uuid-ossp; Type: EXTENSION; Schema: -; Owner: -
+-- Name: make_readable_generic(); Type: FUNCTION; Schema: public; Owner: -
 --
 
-CREATE EXTENSION IF NOT EXISTS "uuid-ossp" WITH SCHEMA public;
+CREATE FUNCTION public.make_readable_generic() RETURNS void
+    LANGUAGE plpgsql
+    SET search_path TO 'public'
+    AS $$
+	declare
+		r record;
+	begin
+		-- do not execute if member of rds_superuser
+		IF EXISTS (select 1 from pg_catalog.pg_roles where rolname = 'rds_superuser')
+		AND pg_has_role(current_user, 'rds_superuser', 'member') THEN
+			RETURN;
+		END IF;
+
+		-- do not execute if superuser
+		IF EXISTS (SELECT 1 FROM pg_user WHERE usename = current_user and usesuper = true) THEN
+			RETURN;
+		END IF;
+
+		-- do not execute if not member of manager role
+		IF NOT pg_has_role(current_user, 'rdsbroker_cbcf231f_9fac_4fa5_8551_85d8d425cf4a_manager', 'member') THEN
+			RETURN;
+		END IF;
+
+		FOR r in (select schema_name from information_schema.schemata) LOOP
+			BEGIN
+				EXECUTE format('GRANT SELECT ON ALL TABLES IN SCHEMA %I TO %I', r.schema_name, 'rdsbroker_cbcf231f_9fac_4fa5_8551_85d8d425cf4a_reader');
+				EXECUTE format('GRANT SELECT ON ALL SEQUENCES IN SCHEMA %I TO %I', r.schema_name, 'rdsbroker_cbcf231f_9fac_4fa5_8551_85d8d425cf4a_reader');
+				EXECUTE format('GRANT USAGE ON SCHEMA %I TO %I', r.schema_name, 'rdsbroker_cbcf231f_9fac_4fa5_8551_85d8d425cf4a_reader');
+
+				RAISE NOTICE 'GRANTED READ ONLY IN SCHEMA %s', r.schema_name;
+			EXCEPTION WHEN OTHERS THEN
+			  -- brrr
+			END;
+		END LOOP;
+
+		RETURN;
+	end
+$$;
 
 
 --
--- Name: EXTENSION "uuid-ossp"; Type: COMMENT; Schema: -; Owner: -
+-- Name: reassign_owned(); Type: FUNCTION; Schema: public; Owner: -
 --
 
-COMMENT ON EXTENSION "uuid-ossp" IS 'generate universally unique identifiers (UUIDs)';
+CREATE FUNCTION public.reassign_owned() RETURNS event_trigger
```

Comment:
These functions and their corresponding event triggers are actually managed by rds and can be safely ignored

Reply:
I'm okay adding them here to just avoid noisy schema changes in future